### ISSUE TITLE
[v2] style tag duplicaton with SSR

### ISSUE THREAD

**Issue by osenvosem** [styled-components/styled-components#512]
SC 2.0.0-4 with babel plugin.

I insert style tag while SSR and on the client it inserts again. Is this normal or a bug?

![screen shot 2017-02-17 at 19 26 21](https://cloud.githubusercontent.com/assets/6762673/23073686/64647c90-f547-11e6-8fbb-057ba021781f.png)

**Comment by nicksrandall** [comment]
I could be wrong, but I *think* this is a glamor problem (which SC uses under-the-hood). Glamor has no way of "rehydrating" itself or even reusing a style tag. I would recommend also filing an issue over there.

**Comment by thysultan** [comment]
@geelen couldn't styled-components add the id/hash to the `<style>` tag i.e `fXgwMh` and do a `getElementById` check on the client side if the style tag has already been inserted with SSR.

**Comment by geelen** [comment]
Yeah except we use more than one style tag per component, and we need to keep a record of the text nodes where we inject stuff so that we can append dynamic styles but keep all CSS for a given component together, otherwise you get nasty ordering bugs. When parsing the existing style tags on app boot, those text nodes don't exist, and so we need a way to serialise the state so that we can simplify rehydration.

Basically, that's all a lot harder than simply rendering the styles twice!

Was talking to @thisguychris about this last month, not sure where he got to.

**Comment by thisguychris** [comment]
@geelen sorry been busy with work.  I'll try to do a WIP this weekend.  I haven't really gone far yet so if anyone wants to do a WIP feel free.

**Comment by mxstbr** [comment]
This is fixed with v2 and @geelen's new stylesheet rehydration stuff!
